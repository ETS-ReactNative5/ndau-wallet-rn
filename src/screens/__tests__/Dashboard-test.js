import React from 'react';
import DashBoard from '../DashBoard';

import renderer from 'react-test-renderer';

describe('testing Dashboard...', () => {
  beforeEach(() => {
    fetch.resetMocks();
  });
  it('renders correctly', () => {
    //Mock out the targetPrice call here
    fetch.mockResponseOnce('12.77');

    const tree = renderer.create(<DashBoard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
