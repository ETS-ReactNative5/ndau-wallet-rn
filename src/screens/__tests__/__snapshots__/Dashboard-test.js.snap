// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Dashboard... renders correctly 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "#1c2227",
      "flex": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#1c2227",
        "flex": 1,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignItems": "center",
              "color": "#ffffff",
              "fontFamily": "TitilliumWeb-Regular",
              "fontSize": 28,
              "justifyContent": "center",
            }
          }
        >
          Wallet 
          7MP-4FV
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignItems": "center",
              "color": "#4d9678",
              "fontFamily": "TitilliumWeb-Regular",
              "fontSize": 48,
              "justifyContent": "center",
              "paddingBottom": 10,
              "textShadowColor": "rgba(77, 150, 120, .5)",
              "textShadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "textShadowRadius": 6,
            }
          }
        >
          NaN
           NDU
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#4d9678",
              "fontFamily": "TitilliumWeb-Regular",
              "fontSize": 20,
              "textShadowColor": "rgba(77, 150, 120, .5)",
              "textShadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "textShadowRadius": 6,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#f75f4b",
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 20,
              }
            }
          >
            **
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 20,
              }
            }
          >
             at current price
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 5,
            "borderWidth": 0.5,
            "height": NaN,
            "marginBottom": 5,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../img/green-locked-card.png",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": undefined,
                  "width": "100%",
                },
                undefined,
              ]
            }
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={Object {}}
          >
            <View
              onLayout={[Function]}
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "flex": 1,
                    "fontFamily": "TitilliumWeb-Bold",
                    "fontSize": 18,
                    "marginBottom": 5,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "marginTop": 5,
                    "textAlign": "left",
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "flex": 1,
                    "fontFamily": "TitilliumWeb-Bold",
                    "fontSize": 18,
                    "marginBottom": 5,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "marginTop": 5,
                    "textAlign": "right",
                  }
                }
              >
                NaN
                 NDU
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "marginBottom": 10,
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
          />
          <View
            onLayout={[Function]}
            style={
              Object {
                "padding": 10,
                "paddingTop": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "TitilliumWeb-Regular",
                  "fontSize": 14,
                }
              }
            >
              This account is not locked
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginBottom": 20,
            "marginRight": 20,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#f75f4b",
              "fontFamily": "TitilliumWeb-Regular",
              "fontSize": 20,
            }
          }
        >
          **
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "TitilliumWeb-Regular",
              "fontSize": 12,
            }
          }
        >
          The estimated value of ndau in US dollars can be calculated using the Target Price at which new ndau have most recently been issued. The value shown here is calculated using that method as of the issue price on 
          09/19/2018
          . The Axiom Foundation bears no responsibility or liability for the calculation of that estimated value, or for decisions based on that estimated value.
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
